hazelcast/src/main/java/com/hazelcast/internal/cluster/fd/PingFailureDetector.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2008-2017, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.cluster.fd;
+
+import com.hazelcast.core.Member;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Ping based failure detector. (OSI Layer 3)
+ * This failure detector uses an absolute number of missing ping attempts.
+ * After that many attempts, a member is considered as dead/unavailable.
+ */
+public class PingFailureDetector implements ClusterFailureDetector {
```

Comment:
It seems `ClusterFailureDetector ` is not a suitable contract for `PingFailureDetector `. Do you still see any value implementing it?

Reply:
while prototyping, it kinda made sense. but now I agree, its better to remove the link between the two and use the concrete type where needed.